From cb8c78b2ffa0b77ae453b2d328d7e2fe5186ef2a Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Tue, 31 Oct 2023 13:32:35 -0300
Subject: linux: Add MMAP_ABOVE4G from Linux 6.6 to sys/mman.h

x86 added the flag (29f890d1050fc099f) for CET enabled.

Also update tst-mman-consts.py test.

Conflicts:
	sysdeps/unix/sysv/linux/tst-mman-consts.py
	  (removed patch to keep alerting about kernel differences)
	sysdeps/unix/sysv/linux/x86/bits/mman.h
	  (adapted to missing __USE_MISC conditional upstream)

diff -rup a/sysdeps/unix/sysv/linux/x86/bits/mman.h b/sysdeps/unix/sysv/linux/x86/bits/mman.h
--- a/sysdeps/unix/sysv/linux/x86/bits/mman.h	2021-08-01 21:33:43.000000000 -0400
+++ b/sysdeps/unix/sysv/linux/x86/bits/mman.h	2024-05-29 14:03:12.426182715 -0400
@@ -26,6 +26,7 @@
 /* Other flags.  */
 #ifdef __USE_MISC
 # define MAP_32BIT	0x40		/* Only give out 32-bit addresses.  */
+# define MAP_ABOVE4G	0x80		/* Only map above 4GB.  */
 #endif
 
 #include <bits/mman-map-flags-generic.h>
